Projects/UOContent/Commands/ClearCommands.cs

```diff
@@ -34,6 +36,58 @@ public static void ClearAll_OnCommand(CommandEventArgs e)
         DeleteObjects(list, from, "globally");
     }
 
+    [Usage("ClearXY x1 y1 x2 y2")]
+    [Description("Deletes all items and mobiles in the specified rectangular area on your facet. Players and their inventory will not be deleted.")]
+    public static void ClearXY_OnCommand(CommandEventArgs e)
+    {
+        if (e.Arguments.Length != 4
+            || !int.TryParse(e.Arguments[0], out var x1)
+            || !int.TryParse(e.Arguments[1], out var y1)
+            || !int.TryParse(e.Arguments[2], out var x2)
+            || !int.TryParse(e.Arguments[3], out var y2))
+        {
+            e.Mobile.SendMessage("Format: ClearXY x1 y1 x2 y2");
+            return;
+        }
+
+        var from = e.Mobile;
+        var map = from.Map;
+
+        if (x2 < x1)
+        {
+            (x1, x2) = (x2, x1);
+        }
+
+        if (y2 < y1)
+        {
+            (y1, y2) = (y2, y1);
+        }
+
+        var rect = new Rectangle2D(x1, y1, x2 - x1 + 1, y2 - y1 + 1);
+        using var queue = PooledRefQueue<IEntity>.Create();
+
+        foreach (var item in map.GetItemsInBounds(rect))
+        {
+            queue.Enqueue(item);
+        }
+
+        foreach (var mob in map.GetMobilesInBounds(rect))
+        {
+            if (!mob.Player)
+            {
+                queue.Enqueue(mob);
+            }
+        }
+
+        var list = new List<IEntity>();
+        while (queue.Count > 0)
+        {
+            list.Add(queue.Dequeue());
+        }
+
+        DeleteObjects(list, from, $"({x1}, {y1}) to ({x2}, {y2}) in {map.Name}");
```

Comment:
I just realized DeleteObjects takes a list, so the queue isn't worth having if we need a list anyways. Sorry about that!